### PR TITLE
Added SignUp page

### LINKED ISSUE #75: Sign-up

### PR DESCRIPTION
**Pull request by mqnguyen5**
## Issue This PR Addresses
Fixes #75 

## Type of Change

- [ ] **Bugfix**: Change which fixes a bug
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

- Added `sign-up` folder and `sign-up/index.js` in `features` folder
- Added sign up route to `page/router/index.js`
- Added link for sign up page navigation to login page
- Added username and confirm password validation to sign up schema
- Added username and confirm password field to sign up form
- Added logic & API calls to complete the sign up workflow

## Steps to test the PR

1. Navigate to the directory and complete the setup for `/web` and `/api` by running the following:

```js
// web
npm run start
// api
npm run dev
```

2. Navigate to `localhost:3000/login`. Inside the page, click on the sign up link
3. When redirected to the new page, fill in all the fields of the sign up form and press "Register"

## Checklist

- [ ] **Test**: make sure this PR passes GitHub action checks
- [ ] **Screenshots**: pictures/vidoes/GIF to demo this PR
- [ ] **Documentation**: this PR includes documentation of new changes, could be comments in code or documentation files

### CODE REVIEW

**Comment by mqnguyen5** [comment]
@TueeNguyen I believe the logic for our register/login process can be refactored further, but at the moment I couldn't come up with any ideas. I suggest we merge this one first after you have completed the code review and testing, then we can discuss on how to refactor this page.

File: src/web/src/features/sign-up/index.js
Comment (TueeNguyen) [inline src/web/src/features/sign-up/index.js:102]: We can change it to 
```suggestion
          email: '',
          username: '',
          password: '',
          confirmPassword: ''
```

File: src/web/src/features/sign-up/index.js
Comment (TueeNguyen) [inline src/web/src/features/sign-up/index.js:58]: I think we can make this config in `axiosInstance` in file `'../../configs/axios'`. You can do it here or in another PR
```suggestion
axiosInstance.defaults.headers.common['Content'] = 'application/json';
```

**Review by TueeNguyen — CHANGES_REQUESTED** [review]

File: src/web/src/features/sign-up/index.js
Comment (mqnguyen5) [reply src/web/src/features/sign-up/index.js:102]: Sounds good. If that's the case, I'll add the placeholder for all of the fields. Do you have any specific placeholder in mind, or should I just do it?

Comment (mqnguyen5) [reply src/web/src/features/sign-up/index.js:102]: Also, if we are setting all the default values to empty strings, we should consider doing this for the login page as well. I can create a small, quick PR for it as a hotfix

File: src/web/src/features/sign-up/index.js
Comment (mqnguyen5) [reply src/web/src/features/sign-up/index.js:58]: Thanks for the info. I'll have a look at it

**Comment by mqnguyen5** [comment]
As for the default values, I replaced them placeholder text for all the fields. Each field has an example so the user knows what field it is. Another approach is to label each of the fields, I'm not too sure about the approach so I'll leave it to your judgement

File: src/web/src/features/sign-up/index.js
Comment (TueeNguyen) [inline src/web/src/features/sign-up/index.js:102]: add `label="Email"` as props for `TextField`

File: src/web/src/features/sign-up/index.js
Comment (TueeNguyen) [inline src/web/src/features/sign-up/index.js:114]: add `label="Username"` as props for `TextField`

File: src/web/src/features/sign-up/index.js
Comment (TueeNguyen) [inline src/web/src/features/sign-up/index.js:126]: add `label="Password"` as props for `TextField`

File: src/web/src/features/sign-up/index.js
Comment (TueeNguyen) [inline src/web/src/features/sign-up/index.js:138]: add `label="Confirm password"` as props for `TextField`

**Review by TueeNguyen — APPROVED** [review]

File: src/web/src/features/sign-up/index.js
Comment (mqnguyen5) [reply src/web/src/features/sign-up/index.js:138]: Got it, I'll add the labels for all the fields

**Comment by TueeNguyen** [comment]
@mqnguyen5, you can add those labels in the textfields as well as the placeholders. After those changes we're GUCCI, let's merge this soon

**Comment by mqnguyen5** [comment]
Understood, will do

**Comment by mqnguyen5** [comment]
Here is the overview of the new changes. I have added the `label` attribute for every text fields inside the sign up page based on Material UI [documentation](https://mui.com/material-ui/react-text-field/#form-props). This should provide the user with not only what each field does but also an example of how the values should be entered


https://user-images.githubusercontent.com/59778930/187050680-05860705-ac69-4e48-ae95-57e92bc1770f.mov

**Review by TueeNguyen — APPROVED** [review]

**Pull request merged**
